File: 13-roman-to-integer/roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        dict={'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        i=0
        res=0
        while(i<len(s)):
            current=dict[s[i]]
            if(i+1<len(s)):
                next=dict[s[i+1]]
                if(current>=next):
                    res+=current
                    i+=1
                else:
                    res+=next-current
                    i+=2
            else:
                res+=current
                i+=1

        return res
        

File: 13-roman-to-integer/roman-to-integer.java
import java.util.*;
class Solution {
    public int romanToInt(String s) {
        /* 
        I=1 
        V=5 
        X=10
        L=50
        C=100
        D=500
        M=1000 
        */
        int res=0;
        HashMap<Character,Integer> map=new HashMap<>();
        map.put('I',1);
        map.put('V',5);
        map.put('X',10);
        map.put('L',50);
        map.put('C',100);
        map.put('D',500);
        map.put('M',1000);
        int i=0;
        while(i<s.length()){
            int current=map.get(s.charAt(i));
            if(i+1<s.length()){
                int next=map.get(s.charAt(i+1));
                if(current>=next){
                    res+=current;
                    i++;
                }else{
                    res=res+next-current;
                    i+=2;
                }
            }else{
                res+=current;
                i++;
            }
        }
        return res;
    }
}